ironmon_tracker/screens/UpdateScreen.lua

```diff
@@ -0,0 +1,309 @@
+	Program.redraw(true)
+
+	if client.GetSoundOn() ~= wasSoundOn then
+		client.SetSoundOn(wasSoundOn)
+	end
+
+	if UpdateScreen.currentState == UpdateScreen.States.SUCCESS then
+		Main.Version.showUpdate = false
+		Main.SaveSettings(true)
+	end
+end
+
+function UpdateScreen.executeBatchOperations()
+	-- For non-Windows OS, likely need to use something other than a .bat file
+	if Main.OS ~= "Windows" then
+		return false
+	end
+
+	-- Temp Files/Folders used by batch operations
+	local archiveName = "Ironmon-Tracker-main.tar.gz"
+	local folderName = "Ironmon-Tracker-main"
+
+	-- Each individual command listed in order, to be appended together later
+	local batchCommands = {
+		'echo Downloading the latest Ironmon Tracker version.',
+		string.format('curl -L "%s" -o "%s"', Constants.Release.TAR_URL, archiveName),
+		'echo Extracting downloaded files.',
+		string.format('tar -xf "%s" && del "%s"', archiveName, archiveName),
+		'echo Applying the update; copying over files.',
+		string.format('rmdir "%s\\.vscode" /s /q', folderName),
+		string.format('rmdir "%s\\ironmon_tracker\\Debug" /s /q', folderName),
+		string.format('del "%s\\.editorconfig" /q', folderName),
+		string.format('del "%s\\.gitattributes" /q', folderName),
+		string.format('del "%s\\.gitignore" /q', folderName),
+		string.format('del "%s\\README.md" /q', folderName),
+		string.format('xcopy "%s" /s /y /q', folderName),
+		string.format('rmdir "%s" /s /q', folderName),
+		'echo Version update completed successfully.',
+		'timeout /t 3',
+	}
+
+	local combined_cmd = ""
+	for _, cmd in ipairs(batchCommands) do
+		combined_cmd = combined_cmd .. cmd .. ' && '
+	end
+	combined_cmd = combined_cmd:sub(1, -5) -- Remove trailing " && "
+
+	print(string.format("Performing version update to %s", Main.Version.latestAvailable))
+
+	local result = os.execute(combined_cmd)
+	if result ~= 0 then -- 0 = successful
+		print("Update-Error: Unable to download, extract, or overwrite files in Tracker folder.")
+		return false
+	end
+
+	print("Update completed successfully.")
+	return true
+end
+
+function UpdateScreen.remindMeLater()
+	Main.Version.remindMe = true
+	Main.Version.showUpdate = false
+	Main.SaveSettings(true)
+	local screenToShow = Utils.inlineIf(Program.isValidMapLocation(), Program.Screens.TRACKER, Program.Screens.STARTUP)
+	Program.changeScreenView(screenToShow)
+end
+
+function UpdateScreen.ignoreTheUpdate()
+	Main.Version.remindMe = false
+	Main.Version.showUpdate = false
+	Main.SaveSettings(true)
+	local screenToShow = Utils.inlineIf(Program.isValidMapLocation(), Program.Screens.TRACKER, Program.Screens.STARTUP)
+	Program.changeScreenView(screenToShow)
+end
+
+function UpdateScreen.openReleaseNotesWindow()
+	-- The first parameter is the title of the window, the second is the url
+	if Main.OS == "Windows" then
+		os.execute(string.format('start "" "%s"', Constants.Release.DOWNLOAD_URL))
+	else
+		os.execute(string.format('open "" "%s"', Constants.Release.DOWNLOAD_URL))
```

Comment:
this won't work on linux, as `os.execute` itself just straight up fails on bizhawk linux annoyingly from what TheChucklingAtom and BurninTheBlues have said in the discord thread

Reply:
Gotcha. I didn't think it would work, but added it in anyway in the event a future Bizhawk version allows this support. I don't have a good way to open a browser window on Linux, nor a Linux system setup to test options.